// @ngrx/store-log-monitor/dock-monitor/index.js
"use strict";
var core_1 = require('@angular/core');
var common_1 = require('@angular/common');
var commander_1 = require('./commander');
var dock_1 = require('./dock');
var dock_monitor_1 = require('./dock-monitor');
var actions_1 = require('./actions');
var DockMonitorModule = (function () {
    function DockMonitorModule() {
    }
    /** @nocollapse */
    DockMonitorModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [
                        common_1.CommonModule
                    ],
                    declarations: [
                        commander_1.CommanderComponent,
                        dock_1.DockComponent,
                        dock_monitor_1.DockMonitorComponent
                    ],
                    providers: [
                        actions_1.DockActions
                    ],
                    exports: [
                        dock_monitor_1.DockMonitorComponent
                    ]
                },] },
    ];
    return DockMonitorModule;
}());
exports.DockMonitorModule = DockMonitorModule;
var reducer_1 = require('./reducer');
exports.useDockMonitor = reducer_1.useDockMonitor;
