// @ngrx/store-log-monitor/store-log-monitor.js
"use strict";
var core_1 = require('@angular/core');
var StoreLogMonitorComponent = (function () {
    function StoreLogMonitorComponent() {
        this.toggleCommand = 'ctrl-h';
        this.positionCommand = 'ctrl-m';
        this.expandEntries = false;
    }
    /** @nocollapse */
    StoreLogMonitorComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'ngrx-store-log-monitor',
                    template: "\n    <dock-monitor [toggleCommand]=\"toggleCommand\" [positionCommand]=\"positionCommand\">\n      <log-monitor [expandEntries]=\"expandEntries\"></log-monitor>\n    </dock-monitor>\n  "
                },] },
    ];
    /** @nocollapse */
    StoreLogMonitorComponent.propDecorators = {
        'toggleCommand': [{ type: core_1.Input },],
        'positionCommand': [{ type: core_1.Input },],
        'expandEntries': [{ type: core_1.Input },],
    };
    return StoreLogMonitorComponent;
}());
exports.StoreLogMonitorComponent = StoreLogMonitorComponent;
