// @ngrx/store-log-monitor/esm/dock-monitor/commander.js
import 'rxjs/add/operator/filter';
import 'rxjs/add/operator/map';
import { Component, Input, Output } from '@angular/core';
import { Subject } from 'rxjs/Subject';
import { KEYCODES } from './keycodes';
export var CommanderComponent = (function () {
    function CommanderComponent() {
        var _this = this;
        this.keydown$ = new Subject();
        this._ignoreTags = ['INPUT', 'SELECT', 'TEXTAREA'];
        this.command = this.keydown$
            .filter(function (e) { return _this._ignoreTags.indexOf(e.target.tagName) < 0; })
            .filter(function (e) { return !(e.target.isContentEditable); })
            .filter(function (e) {
            var command = _this.parseCommand(_this.shortcut);
            if (!command) {
                return false;
            }
            var charCode = e.keyCode || e.which;
            var char = String.fromCharCode(charCode);
            return command.name.toUpperCase() === char.toUpperCase() &&
                command.alt === e.altKey &&
                command.ctrl === e.ctrlKey &&
                command.meta === e.metaKey &&
                command.shift === e.shiftKey;
        })
            .map(function (e) {
            e.preventDefault();
            return { command: _this.shortcut };
        });
    }
    CommanderComponent.prototype.parseCommand = function (s) {
        var keyString = s.trim().toLowerCase();
        if (!/^(ctrl-|shift-|alt-|meta-){0,4}\w+$/.test(keyString)) {
            throw new Error('The string to parse needs to be of the format "c", "ctrl-c", "shift-ctrl-c".');
        }
        var parts = keyString.split('-');
        var key = {
            ctrl: false,
            meta: false,
            shift: false,
            alt: false
        };
        var c;
        key.name = parts.pop();
        while ((c = parts.pop())) {
            key[c] = true;
        }
        if (key.ctrl) {
            key.sequence = KEYCODES.ctrl[key.name] || key.name;
        }
        else {
            key.sequence = KEYCODES.nomod[key.name] || key.name;
        }
        if (key.shift && key.sequence && key.sequence.length === 1) {
            key.sequence = key.sequence.toUpperCase();
        }
        return key;
    };
    /** @nocollapse */
    CommanderComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngrx-commander',
                    template: '',
                    styles: [':host{ display: none }'],
                    host: {
                        '(document:keydown)': 'keydown$.next($event)'
                    }
                },] },
    ];
    /** @nocollapse */
    CommanderComponent.propDecorators = {
        'shortcut': [{ type: Input },],
        'command': [{ type: Output },],
    };
    return CommanderComponent;
}());
